src/Desktop/ArcGISRuntime.Desktop.Samples/Samples/Data/FeatureLayerQuery/FeatureLayerQuery.xaml.cs

```diff
@@ -0,0 +1,128 @@
+// language governing permissions and limitations under the License.
+
+using Esri.ArcGISRuntime.Data;
+using Esri.ArcGISRuntime.Geometry;
+using Esri.ArcGISRuntime.Mapping;
+using Esri.ArcGISRuntime.Symbology;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+
+namespace ArcGISRuntime.Desktop.Samples.FeatureLayerQuery
+{
+    /// <summary>
+    /// Interaction logic for FeatureLayerQuery.xaml
+    /// </summary>
+    public partial class FeatureLayerQuery : UserControl
+    {
+
+        // Create reference to service of US States  
+        private string _statesUrl = "http://sampleserver6.arcgisonline.com/arcgis/rest/services/USA/MapServer/2";
+
+        // Create globally available feature table for easy referencing 
+        private ServiceFeatureTable _featureTable;
+
+        // Create globally available feature layer for easy referencing 
+        private FeatureLayer _featureLayer;
+
+        public FeatureLayerQuery()
+        {
+            InitializeComponent();
+
+            // Create the UI, setup the control references and execute initialization 
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            // Create new Map with basemap
+            Map myMap = new Map(Basemap.CreateTopographic());
+
+            // Create and set initial map location
+            Esri.ArcGISRuntime.Geometry.MapPoint initialLocation = new MapPoint(
+                -11000000, 5000000, SpatialReferences.WebMercator);
+            myMap.InitialViewpoint = new Viewpoint(initialLocation, 100000000);
+
+            // Create feature table using a url
+            _featureTable = new ServiceFeatureTable(new Uri(_statesUrl));
+
+            // Create feature layer using this feature table
+            _featureLayer = new FeatureLayer(_featureTable);
+
+            // Set the Opacity of the Feature Layer
+            _featureLayer.Opacity = 0.6;
+
+            // Create a new renderer for the States Feature Layer
+            SimpleLineSymbol lineSymbol = new SimpleLineSymbol(
+                SimpleLineSymbolStyle.Solid, Colors.Black, 1);
+            SimpleFillSymbol fillSymbol = new SimpleFillSymbol(
+                SimpleFillSymbolStyle.Solid, Colors.Yellow, lineSymbol);
+
+            // Set States feature layer renderer
+            _featureLayer.Renderer = new SimpleRenderer(fillSymbol);
+
+            // Add feature layer to the map
+            myMap.OperationalLayers.Add(_featureLayer);
+
+            // Assign the map to the MapView
+            myMapView.Map = myMap;
+        }
+
+        private async void OnQueryClicked(object sender, RoutedEventArgs e)
+        {
+            // Remove any previous feature selections that may have been made 
+            _featureLayer.ClearSelection();
+
+            // Begin query process 
```

Comment:
I think that begin is a wrong term here since we are starting the query and waiting that it is finished.